packages/client-assertion-validation/test/validation.test.ts

```diff
@@ -0,0 +1,878 @@
+      expect(errors).toBeDefined();
+      expect(errors).toHaveLength(2);
+      expect(errors).toEqual([jtiNotFound(), issuerNotFound()]);
+    });
+
+    it("subjectNotFound", () => {
+      const jws = getMockClientAssertion({
+        customHeader: {},
+        standardClaimsOverride: { sub: undefined },
+        customClaims: {},
+      });
+      const { errors } = verifyClientAssertion(jws, undefined);
+      expect(errors).toBeDefined();
+      expect(errors).toHaveLength(1);
+      expect(errors![0]).toEqual(subjectNotFound());
+    });
+
+    it("invalidSubject - Subject claim differs from clientID parameter", () => {
+      const subject = generateId<ClientId>();
+      const jws = getMockClientAssertion({
+        customHeader: {},
+        standardClaimsOverride: { sub: subject },
+        customClaims: {},
+      });
+      const { errors } = verifyClientAssertion(jws, generateId<ClientId>());
+      expect(errors).toBeDefined();
+      expect(errors).toHaveLength(1);
+      expect(errors![0]).toEqual(invalidSubject(subject));
+    });
+
+    it("invalidSubjectFormat", () => {
+      const clientId: ClientId = generateId();
+      const subject = "not a client id";
+      const jws = getMockClientAssertion({
+        customHeader: {},
+        standardClaimsOverride: { sub: subject },
+        customClaims: {},
+      });
+      const { errors } = verifyClientAssertion(jws, clientId);
+      expect(errors).toBeDefined();
+      expect(errors).toHaveLength(1);
+      expect(errors![0]).toEqual(invalidSubjectFormat(subject));
+    });
+
+    it("invalidPurposeIdClaimFormat", () => {
+      const notPurposeId = "not a purpose id";
+      const jws = getMockClientAssertion({
+        customHeader: {},
+        standardClaimsOverride: {},
+        customClaims: {
+          purposeId: notPurposeId,
+        },
+      });
+      const { errors } = verifyClientAssertion(jws, undefined);
+      expect(errors).toBeDefined();
+      expect(errors).toHaveLength(1);
+      expect(errors![0]).toEqual(invalidPurposeIdClaimFormat(notPurposeId));
+    });
+
+    it("invalidClientIdFormat", () => {
+      const notClientId = "not a client id";
+      const jws = getMockClientAssertion({
+        customHeader: {},
+        standardClaimsOverride: {},
+        customClaims: {},
+      });
+      const { errors } = verifyClientAssertion(jws, notClientId);
+      expect(errors).toBeDefined();
+      expect(errors).toHaveLength(1);
+      expect(errors![0]).toEqual(invalidClientIdFormat(notClientId));
+    });
+
+    it("should not throw error if digest is undefined", () => {
+      const jws = getMockClientAssertion({
+        customHeader: {},
+        standardClaimsOverride: {},
+        customClaims: {
+          digest: undefined,
+        },
+      });
+
+      const verifiedClientAssertion = verifyClientAssertion(jws, undefined);
+      expect(verifiedClientAssertion.data?.payload.digest).toBeUndefined();
+    });
```

Comment:
@galales this test has been updated to address [this](https://github.com/pagopa/interop-be-monorepo/pull/956#discussion_r1782279492) comment

Reply:
perfect, thanks!